x/ecocredit/server/marketplace/prune.go

```diff
@@ -24,16 +24,26 @@ func (k Keeper) PruneSellOrders(ctx context.Context) error {
 		return err
 	}
 	defer it.Close()
+
+	expiredOrderIds := make([]uint64, 0)
 	for it.Next() {
 		sellOrder, err := it.Value()
 		if err != nil {
 			return err
 		}
-		if err = k.unescrowCredits(ctx, sellOrder.Seller, sellOrder.BatchId, sellOrder.Quantity); err != nil {
+		if sellOrder.Expiration != nil {
+			if err = k.unescrowCredits(ctx, sellOrder.Seller, sellOrder.BatchId, sellOrder.Quantity); err != nil {
+				return err
+			}
+			expiredOrderIds = append(expiredOrderIds, sellOrder.Id)
+		}
+	}
+	for _, id := range expiredOrderIds {
+		if err := k.stateStore.SellOrderTable().Delete(ctx, &marketplaceapi.SellOrder{Id: id}); err != nil {
```

Comment:
Actually, it looks like we can call this within the `it.Next()` loop rather than appending to `expiredOrderIds`. No need to create `expiredOrderIds` and append the expired orders and then loop through a second time.

Reply:
per https://github.com/cosmos/cosmos-sdk/blob/main/orm/model/ormtable/iterator.go#L15-L19 we should not mutate the table during iteration. 

however, i came back to this and actually might have a better solution 😈 . so after talking with aaron, nil timestamps are actually treated as zero values, or `timestamppb.Timestamp{Seconds:0, Nanos: 0}`. so if we want the performance boost of `DeleteRange`, we could just delete everything starting at `timestamppb.Timestamp{Seconds:0, Nanos: 1}`. 

the tests still hold up with this [change](https://github.com/regen-network/regen-ledger/pull/1115/commits/f60f734d5843e3f7224ba960390eefdebc55229a) 😄 . lmk if that sounds sane or not